### DISCUSSION TITLE
Hello, I would like to adjust the Hounsfield Unit (HU) values of generated X-ray images. Which interface should I call for this purpose?

### DISCUSSION THREAD

**Discussion by bjy12** [CERN/TIGRE#510] [Q&A]
I have generated numerous CBCT (Cone Beam Computed Tomography) images of knee joints based on examples in Python. However, the skin part in the images is too prominent. I would like to make the skin less visible by adjusting the Hounsfield Unit (HU) values.
"I have generated knee joint images as follows:"
![image](https://github.com/CERN/TIGRE/assets/99953398/0caaa2a0-59de-4dbd-9069-5fac56adb693)
"I would like to generate knee joint images similar to the following:"
![image](https://github.com/CERN/TIGRE/assets/99953398/4da5bebf-7eb2-4f1e-a05b-f114a6a02944)
Furthermore, I sincerely appreciate your provision of such a powerful open-source tool for us

**Comment by AnderBiguri** [comment]
Hi @bjy12 I am not sure what are you having trouble with, can you explain better?

TIGRE does not work in HUs, but linear attenuation coefficients. Additionally, projections are not in HUs, ever. 
But also, if you want a different attenuation, you can change the image you used to generate them. Finally, maybe you just want to change the visualization range, the `clims`?

**Comment by bjy12** [comment]
I apologize
I didn't clearly describe the issue I'm facing.
I have images using TIGRE's DRR, and they look like this
[image: image.png]
And the raw CT image  is look like this
[image: image.png]
 But I want to generate DRR  image like this
[image: image.png]
I want the generated DRR images to have clearer bone structures, and I
would like the grayscale representation of the skin to be lower.

Biguri ***@***.***> 于2023年12月4日周一 16:28写道：

**Reply by AnderBiguri** [reply]
I can not see the images either, please post them on Github, don't reply to the email

**Comment by tsadakane** [comment]
Hi, @bjy12 
Although I cannot see the images at https://github.com/CERN/TIGRE/discussions/510#discussioncomment-7760991 , I suppose that the first image is a forward projection image generated by TIGRE, the second image is a sagittal or coronal image of TIGRE's CT, and the third is the second image in [the first comment](https://github.com/CERN/TIGRE/discussions/510#discussion-5921051). 
I think you need to know more about how the third image was generated and then simulate it.  For the simulation, you can use the TIGRE's forward projection image(s) as the RAW data of the DRR.

**Comment by bjy12** [comment]
Thank you for your response. In fact, I would like to enhance the contrast
between the skin and bones in the generated DRR images.

**Comment by AnderBiguri** [comment]
I think my first commetn still stands. If you generate DDRs with TIGRE (knowing that they simulate monochromatic non-scattered perfect X-rays), there is no other way to generate a different looking image. There is only 1 radiograph possible existing for a given geometry.

There are 2 things that when changed will generate a different radiograph:
 1- change the raw CT image (this is not a TIGRE thing, just change the image). 
2-Change the physics simulation (which you can not do in TIGRE). 

However, I still just wonder if you are having issues with visualization. What exact code did you use to visualize the radiograph? Because if you change the limits, the bone will look whiter. Just make sure you are not confusing visualization range with data value.

**Comment by bjy12** [comment]
Thank you once again for your response. Currently, I am reading the
original CT images using SimpleITK and converting them into arrays,
followed by normalization. Could you please clarify what you mean by
changing the original CT images? Additionally, the code I use for
visualizing DRR images is tigre.plotproj(cfg['train']['projections']).

Biguri ***@***.***> 于2023年12月5日周二 19:27写道：

**Reply by AnderBiguri** [reply]
I meant literally, have a different image, then it will provide a different DDR. 

try visualizing with

```
max_val=1000
tigre.plotproj(cfg['train']['projections'], clims=[0, max_val]).

```
And changing the value of `max_val` to see if that solves your issue.

**Comment by bjy12** [comment]
Thank you for your clarification. What I meant is that before using TIGRE
to generate DRR (Digitally Reconstructed Radiograph) based on the original
CT, I normalized the original image. I used simpleITK to read the CT images
and obtained them in array form. I then performed normalization on the
array representing the CT images. Will this affect the quality of DRR
generation, or should I refrain from normalization?



Biguri ***@***.***> 于2023年12月5日周二 20:08写道：

**Reply by AnderBiguri** [reply]
I don't know what ITK does for normalization, but if what it does change the images to [0-1] range, then its fine. The only thing is that your DDR will ahve a different range.  I suggest replying in Github directly rather than in the automatic emails, as Github supports conversations

**Comment by bjy12** [comment]
Ok, thanks.

Biguri ***@***.***> 于2023年12月6日周三 21:59写道：

**Reply by bjy12** [reply]
You means i should changes the raw ct image to [0-1] range. It is more safely.

**Reply by AnderBiguri** [reply]
No, I don't. I mean that's perhaps what you have already done.... You said you normalised.
